### PR TITLE
Arachne 1995/multi atlas

### PR DESCRIPTION
**Pull request by pavgra**

### CODE REVIEW

File: src/main/java/com/odysseusinc/arachne/portal/api/v1/controller/BaseDataNodeMessagingController.java [hunk 1]
```diff
@@ -129,19 +133,35 @@ public void saveListResponse(
...
-        }
-        if (!Objects.equals(datanode.getAtlasVersion(), atlasVersion)) {
-            datanode.setAtlasVersion(atlasVersion);
-            baseDataNodeService.updateAtlasInfo(datanode);
+        A atlas = convert(atlasShortDTO);
+        A updated;
+
+        if (atlas.getId() == null || atlasService.findByIdInAnyTenant(atlas.getId()) == null) {
+            final DN dataNode = getDatanode(principal);
+
```
Comment (wivern) [inline src/main/java/com/odysseusinc/arachne/portal/api/v1/controller/BaseDataNodeMessagingController.java]: Just to clarify, if Atlas exists but not assigned to any tenant one would be saved twice. Is that expected behavior?

File: src/main/java/com/odysseusinc/arachne/portal/model/BaseAtlas.java [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+
+        BaseAtlas atlas = (BaseAtlas) o;
+
+        return getId() != null ? getId().equals(atlas.getId()) : atlas.getId() == null;
+    }
+
+    @Override
+    public int hashCode() {
+
+        return getId() != null ? getId().hashCode() : 0;
```
Comment (agackovka) [inline src/main/java/com/odysseusinc/arachne/portal/model/BaseAtlas.java]: 0? why Object.hashCode() doesn't fit?

File: src/main/java/com/odysseusinc/arachne/portal/model/DataNode.java [hunk 3]
```diff
@@ -91,15 +91,25 @@
...
     @Transient
     private Set<ArachnePermission> permissions;
 
     @Column
     private Boolean published;
 
+    public String getAtlasVersion() {
+
+        return this.getAtlasList().stream()
+                .filter(a -> a.getVersion() != null)
```
Comment (agackovka) [inline src/main/java/com/odysseusinc/arachne/portal/model/DataNode.java]: can be replaced by Objects.nonNull

Comment (pavgra) [reply src/main/java/com/odysseusinc/arachne/portal/model/DataNode.java]: It already contains "Objects" from Google package, so...

File: src/main/java/com/odysseusinc/arachne/portal/api/v1/controller/BaseDataNodeMessagingController.java (hunk 1 above)
Comment (pavgra) [reply src/main/java/com/odysseusinc/arachne/portal/api/v1/controller/BaseDataNodeMessagingController.java]: I've added Javadoc for clariication of method naming

**Pull request merged**